swift/ql/src/queries/Security/CWE-757/TLS13IsNotUsed.ql

```diff
@@ -0,0 +1,58 @@
+/**
+ * @name TODO
+ * @description TODO
+ * @kind path-problem
+ * @problem.severity error
+ * @security-severity TODO
+ * @precision high
+ * @id swift/tls13-is-not-used
+ * @tags security
+ *       external/cwe/cwe-757
+ */
+
+import swift
+import codeql.swift.dataflow.DataFlow
+import codeql.swift.dataflow.TaintTracking
+import codeql.swift.dataflow.FlowSources
+import DataFlow::PathGraph
+
+/**
+ * TODO
+ */
+class TLS13IsNotUsedConfig extends TaintTracking::Configuration {
+  TLS13IsNotUsedConfig() { this = "TLS13IsNotUsedConfig" }
+
+  override predicate isSink(DataFlow::Node node) {
+    exists(AssignExpr assign, MemberRefExpr member, string memberName |
+      assign.getSource() = node.asExpr() and
```

Comment:
Currently, this will only be seen as a sink if the write to the configuration is through an `AssignExpr`. If we want to generalize this we could let the sink be `sink.(DataFlow::PostUpdateNode).getPreUpdateNode().asExpr() = member.getBase()` (or whatever we call `a` in `a.b`). This would mean we recognized the write to the config no matter where it came from (i.e., in case it doesn't come from an `AssignExpr`).

However, we'd also need to be careful about the fields involved. We might need to also register `sink.(DataFlow::PostUpdateNode).getPreUpdateNode()` as implicitly reading the fields enumerated by `memberName` in `isSink` 🤔.

TLDR: In any case, this is good enough for now! But we can probably make this a bit less syntax-dependent in a future PR.

Reply:
OOC, what else could it be if not assignment? 🤔

Reply:
I'm still not totally familiar with the Swift AST, but in CPP initializers are not assignment for example.

Reply:
maybe a write coming out a function like:
```swift
func set(_ protocol : inout SSLProtocol) {
  procotol = SSLProtocol.tlsProtocol10
}

func test() {
  let config = URLSessionConfiguration()
  set(&config.tlsMaximumSupportedProtocol); // BAD
}
```

(Assuming that compiles, obviously 😂)